waiter/src/waiter/process_request.clj

```diff
@@ -485,13 +498,22 @@
 
 (defn abort-http-request-callback-factory
   "Creates a callback to abort the http request."
-  [response]
-  (fn abort-http-request-callback [^Exception e]
-    (let [ex (if (instance? IOException e) e (IOException. e))
-          aborted (if-let [request (:request response)]
-                    (.abort request ex)
-                    (log/warn "unable to abort as request not found inside response!"))]
-      (log/info "aborted backend request:" aborted))))
+  [{:keys [abort-ch] :as response}]
+  (fn abort-backend-request-callback [^Exception ex]
+    (let [correlation-id (cid/get-correlation-id)
+          callback (fn abort-http-request-result-callback [aborted?]
+                     (cid/with-correlation-id
+                       correlation-id
+                       (log/info ex "aborted backend request:" aborted?)))]
+      (when-not abort-ch
+        (log/warn "abort-ch not available in response"))
+      (log/debug "aborting backend request")
+      (when-not (and abort-ch (async/put! abort-ch [ex callback]))
+        (log/debug "aborting backend request directly")
```

Comment:
If the `abort-ch` is closed, doesn't that mean the backend request is already aborted?

Reply:
The branch is also covering the case when the `abort-ch` is not available.

`abort-ch` closed means we do not have a way to asynchronously triggering the abort logic. Currently, `abort-ch` is only closed when we are done processing the request:
https://github.com/twosigma/jet/blob/e1f0a7b279c564c4a457ee1d020797c818bc7f39/src/clj/qbits/jet/client/http.clj#L411